core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -385,6 +368,141 @@ sealed abstract class MultiSigSigner
+
+          val hashForSig = TransactionSignatureSerializer.hashForSignature(
+            txSigComponent = wtxSigComp,
+            hashType = spendingInfo.hashType)
+
+          //sign the hash
+          val signature: ECDigitalSignature = {
+            if (isDummySignature) {
+              EmptyDigitalSignature
+            } else {
+              val sig = spendingInfo.signers.head.sign(hashForSig.bytes)
+              //append hash type
+              ECDigitalSignature.fromBytes(
+                sig.bytes.:+(spendingInfo.hashType.byte))
+            }
+          }
+          val p2wpkhWit =
+            P2WPKHWitnessV0(spendingInfo.signers.head.publicKey, signature)
+          val updatedWit = uwtx.updateWitness(inputIndex.toInt, p2wpkhWit)
+
+          Future.successful(
+            P2SHTxSigComponent(updatedWit, inputIndex, output, flags))
+        case (nestedSpendingInfo: P2SHNestedSegwitV0UTXOSpendingInfo,
+              _: P2WSHWitnessSPKV0) =>
+          val outpoint = spendingInfo.outPoint
+          val redeemScript = nestedSpendingInfo.redeemScript
+          val oldInput = unsignedTx.inputs(inputIndex.toInt)
+
+          val input =
+            TransactionInput(outpoint, EmptyScriptSignature, oldInput.sequence)
+
+          val updatedTx =
+            unsignedTx.updateInput(inputIndex.toInt, input)
+
+          val updatedOutput =
+            TransactionOutput(spendingInfo.amount, redeemScript)
+
+          val updatedUTXOInfo = BitcoinUTXOSpendingInfo(
+            outpoint,
+            updatedOutput,
+            spendingInfo.signers,
+            None,
+            spendingInfo.scriptWitnessOpt,
+            spendingInfo.hashType,
+            spendingInfo.conditionalPath)
+
+          val signedTxEither =
+            BitcoinSigner
+              .sign(updatedUTXOInfo, updatedTx, isDummySignature)
+              .map(_.transaction)
+
+          signedTxEither.map { signedTx =>
+            val i = signedTx.inputs(inputIndex.toInt)
+
+            val p2sh =
+              P2SHScriptSignature(i.scriptSignature, redeemScript)
+
+            val signedInput =
+              TransactionInput(i.previousOutput, p2sh, i.sequence)
+
+            val signedInputs =
+              signedTx.inputs.updated(inputIndex.toInt, signedInput)
+
+            val finalTx = signedTx match {
+              case btx: BaseTransaction =>
+                BaseTransaction(btx.version,
+                                signedInputs,
+                                btx.outputs,
+                                btx.lockTime)
+              case wtx: WitnessTransaction =>
+                WitnessTransaction(wtx.version,
+                                   signedInputs,
+                                   wtx.outputs,
+                                   wtx.lockTime,
+                                   wtx.witness)
+            }
+            P2SHTxSigComponent(finalTx, inputIndex, output, flags)
+          }
+        case (_, _) =>
```

Comment:
In general it is a bad idea to use `_` in a pattern match. 

For one, the first `(_,` can never be anything, and the second `_` can be typed with a `_: RawScriptPubKey` if I understand this. You probably should just pattern match on `spendingInfoToSatisfy` and then inside of that match on `spendingInfoToSatisfy.redeemScript`